src/commands/test/test_environment_exceptions.py

```diff
@@ -1,61 +1,118 @@
-from typing import Optional
-
-from starkware.starkware_utils.error_handling import StarkException
-
-from src.commands.test.utils import extract_core_info_from_stark_ex_message
+from typing import List, Optional
 
 
 class ReportedException(BaseException):
+    """
+    The exception used for catching unexpected errors thrown from test cases and as a base class.
+    """
+
     def __str__(self) -> str:
         return str(super().__repr__())
 
 
-class MissingExceptException(ReportedException):
-    pass
+class RevertableException(BaseException):
+    """
+    This exception is used by `except_revert` logic.
+    """
+
+    def __init__(
+        self,
+        error_message: Optional[str] = None,
+        error_type: Optional[str] = None,
+    ) -> None:
+        super().__init__(error_message)
+        self.error_type = error_type
+        self.error_message = error_message
+
+    def __str__(self) -> str:
+        result: List[str] = []
+        if self.error_type is not None:
+            result.append(f"[error_type] {self.error_type}")
 
+        if self.error_type is not None:
+            result.append(f"[error_message] {self.error_message}")
+        return "\n".join(result)
+
+    def match(self, other: "RevertableException") -> bool:
+        return (self.error_type is None or self.error_type == other.error_type) and (
+            self.error_message is None
+            or self.error_message in (other.error_message or "")
+        )
+
+
+class StandardRevertableException(RevertableException):
+    """
+    The exception commonly used to display errors encountered during test execution.
```

Comment:
We should mention here it's an abstraction over all errors raised in cairo runtime